server/src-lib/Hasura/GraphQL/Resolve/Select.hs

```diff
@@ -95,13 +95,11 @@ fromAggSelSet
      )
   => G.NamedType -> SelSet -> m TableAggFlds
 fromAggSelSet fldTy selSet = fmap toFields $
-  withSelSet selSet $ \f -> do
-    let fTy = _fType f
-        fSelSet = _fSelSet f
-    case _fName f of
+  withSelSet selSet $ \Field{..} -> 
+    case _fName of
       "__typename" -> return $ RS.TAFExp $ G.unName $ G.unNamedType fldTy
-      "aggregate"  -> RS.TAFAgg <$> convertAggFld fTy fSelSet
-      "nodes"      -> RS.TAFNodes <$> fromSelSet fTy fSelSet
+      "aggregate"  -> RS.TAFAgg <$> convertAggFld _fType _fSelSet
```

Comment:
is it important to change these functions? Seems unrelated to remote joins?

Reply:
That's to squash shadowing errors that came up when I derived lenses for `Field` unfortunately